pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,774 @@
+	// Incorrect settings on this feild maybe lead to the corresponding component in an unhealthy
+	// state. Before you do it, please confirm that you understand the risks of this configuration.
+	//
+	// For supported flags, please see
+	// https://github.com/karmada-io/karmada/blob/master/cmd/controller-manager/app/options/options.go
+	// for details.
+	// +optional
+	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
+
+	// FeatureGates enabled by the user.
+	// - Failover: https://karmada.io/docs/userguide/failover/#failover
+	// - GracefulEviction: https://karmada.io/docs/userguide/failover/#graceful-eviction-feature
+	// - PropagateDeps: https://karmada.io/docs/userguide/scheduling/propagate-dependencies
+	// - CustomizedClusterResourceModeling: https://karmada.io/docs/userguide/scheduling/cluster-resources#start-to-use-cluster-resource-models
+	// More info: https://github.com/karmada-io/karmada/blob/master/pkg/features/features.go
+	// +optional
+	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+}
+
+// KarmadaScheduler holds settings to karmada-scheduler component of the karmada.
+type KarmadaScheduler struct {
+	// CommonSettings holds common settings to karmada scheduler.
+	CommonSettings `json:",inline"`
+
+	// ExtraArgs is an extra set of flags to pass to the karmada-scheduler component or override.
+	// A key in this map is the flag name as it appears on the command line except without
+	// leading dash(es).
+	//
+	// Note: This is a temporary solution to allow for the configuration of the karmada-scheduler
+	// component. In the future, we will provide a more structured way to configure the component.
+	// Once that is done, this field will be discouraged to be used.
+	// Incorrect settings on this feild maybe lead to the corresponding component in an unhealthy
+	// state. Before you do it, please confirm that you understand the risks of this configuration.
+	//
+	// For supported flags, please see
+	// https://github.com/karmada-io/karmada/blob/master/cmd/scheduler/app/options/options.go
+	// for details.
+	// +optional
+	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
+
+	// FeatureGates enabled by the user.
+	// - CustomizedClusterResourceModeling: https://karmada.io/docs/userguide/scheduling/cluster-resources#start-to-use-cluster-resource-models
+	// More info: https://github.com/karmada-io/karmada/blob/master/pkg/features/features.go
+	// +optional
+	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+}
+
+// KarmadaDescheduler holds settings to karmada-descheduler component of the karmada.
+type KarmadaDescheduler struct {
+	// CommonSettings holds common settings to karmada descheduler.
+	CommonSettings `json:",inline"`
+
+	// Enable indicates whether the karmada-descheduler component should be deployed.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// Defaults to false.
+	// +optional
+	Enable *bool `json:"enable,omitempty"`
```

Comment:
The other questions is about the field `Enable`,  whether all components require this field?

Reply:
Good question. I think we can remove it from all components.
If `.spec.components.xxx == nil` means disabled, otherwise means enabled.

Reply:
Agree with it.